client/src/components/molecules/ClientRequestTable.tsx

```diff
@@ -0,0 +1,229 @@
+    mutation updateRequest($request: RequestInput) {
+      updateRequest(request: $request) {
+        id
+        success
+        message
+      }
+    }
+  `;
+  const softDeleteRequest = gql`
+    mutation deleteRequest($id: ID) {
+      softDeleteRequest(id: $id) {
+        id
+        success
+        message
+      }
+    }
+  `;
+
+  const [requests, setRequests] = useState<Request[]>([]);
+  useEffect(() => {
+    const requests = props.requests.filter((request) => !request.deleted);
+    const nonFulfilledRequests = requests.filter((request) => {
+      if (request !== undefined) {
+        if (request.fulfilled === false) {
+          return request;
+        }
+      }
+    });
+    const fulfilledRequests = requests.filter((request) => {
+      if (request !== undefined) {
+        if (request.fulfilled === true) {
+          return request;
+        }
+      }
+    });
+
+    nonFulfilledRequests.sort((a, b) => {
+      return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+    });
+
+    fulfilledRequests.sort((a, b) => {
+      return a!.dateCreated!.valueOf() - b!.dateCreated!.valueOf();
+    });
+
+    const sortedRequests: Request[] = nonFulfilledRequests!.concat(
+      fulfilledRequests!
+    ) as Request[];
+    setRequests(sortedRequests);
+  }, [props.requests]);
+
+  const [mutateDeleteRequest] = useMutation(softDeleteRequest);
+  const [mutateRequest] = useMutation(updateRequest);
+
+  const onSoftDeleteRequest = (index: number) => {
+    const requestsCopy = requests.slice();
+    const req = { ...requestsCopy[index] };
+    requestsCopy.splice(index, 1);
+    const id = req._id;
+    setRequests(requestsCopy);
+    props.onChangeRequests!(requestsCopy);
+    mutateDeleteRequest({ variables: { id: id } });
+  };
+
+  const onFulfilledRequest = (index: number) => {
+    const requestsCopy = requests.slice();
+    const req = { ...requestsCopy[index] };
+    if (req.fulfilled === false) {
+      req.fulfilled = true;
+      requestsCopy.splice(index, 1);
+      let i = requestsCopy.length - 1;
+      for (; i > -1; --i) {
+        if (requestsCopy[i].fulfilled === false) break;
+        else if (
+          requestsCopy[i]!.dateCreated!.valueOf() < req!.dateCreated!.valueOf()
+        )
+          break;
+      }
+      requestsCopy.splice(i + 1, 0, req);
+    } else {
+      req.fulfilled = false;
+      requestsCopy.splice(index, 1);
+      let i = 0;
+      for (; i < requestsCopy.length; ++i) {
+        if (requestsCopy[i].fulfilled === true) break;
+        else if (
+          requestsCopy[i]!.dateCreated!.valueOf() > req!.dateCreated!.valueOf()
+        )
+          break;
+      }
+      requestsCopy.splice(i, 0, req);
+    }
+    setRequests(requestsCopy);
+    props.onChangeRequests!(requestsCopy);
+    const id = req._id;
+    const requestId = req.requestId;
+    const fulfilled = req.fulfilled;
+    mutateRequest({ variables: { request: { id, requestId, fulfilled } } });
```

Comment:
I'll remove `requestId` if it isn't needed in the mutation request. 

For the variables, I generally prefer doing destructing it like
```
const {_id: id, requestId, fulfilled} = req;
mutateRequest({ variables: { request: { id, requestId, fulfilled } } });
```
over something like
```
mutateRequest({ variables: { request: { id: req._id, requestId: req.requestId, fulfilled: req.fulfilled } } });
```
what are your thoughts?

Reply:
`requestId` can be removed since it's just a human-readable id for a request ([documented here](https://www.notion.so/uwblueprintexecs/MongoDB-Schemas-577ae0f4beb34778b754b56853cdbbad)). We use `_id` to uniquely identify all objects (include request objects)

Reply:
Personally, I think option 2 is fine @Puepis as long as it's formatted on multiple lines
```
mutateRequest({
  variables: {
    request: {
      id: req._id,
      requestId: req.requestId,
      fulfilled: req.fulfilled
    }
  }
});
```

Reply:
done!